plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/LdapAuthenticator.java

```diff
@@ -81,6 +84,9 @@ public LdapAuthenticator(LdapAuthenticatorClient client, LdapConfig ldapConfig)
         checkArgument(
                 bindDistinguishedName.isPresent() || !userBindSearchPatterns.isEmpty(),
                 "Either user bind search pattern or bind distinguished name must be provided");
+        checkArgument(
```

Comment:
Can you please provide tests for this validation?

Reply:
Can we have this assertion as a part of the `LdapConfig` ? And annotate with `AssertTrue` annotation so that when building config object.

Reply:
> Can you please provide tests for this validation?

If ignore certificate, I think that trust certificate option is useless. And two configuration cannot use same sslcontext.

Reply:
It is useless. So we need to make sure that user can only use one of the option as using them both is very likely a misconfiguration.

Reply:
> Can we have this assertion as a part of the `LdapConfig` ? And annotate with `AssertTrue` annotation so that when building config object.

I tried to move this logic to LdapConfig, like your opinion but I met huddle. When I tested with `testExplicitConfig` method in TestLdapConfig, this method look like all configurations must be set but this logic, only one configuration can be set. How to fixed this situation?

Reply:
In case of `testExplicitConfig` we could set `ignoreCertificate` to false and have an additional tests that check for the invalid condition.